admiral/pkg/clusters/serviceentry.go

```diff
@@ -523,7 +525,16 @@ func AddServiceEntriesWithDr(ctx context.Context, rr *RemoteRegistry, sourceClus
 						//nolint
 						newServiceEntry := createServiceEntrySkeletion(*seDr.ServiceEntry, seDr.SeName, syncNamespace)
 						if newServiceEntry != nil {
-							newServiceEntry.Labels = map[string]string{common.GetWorkloadIdentifier(): fmt.Sprintf("%v", identityId)}
+							newServiceEntry.Labels = map[string]string{
+								common.GetWorkloadIdentifier(): fmt.Sprintf("%v", identityId),
+								common.GetEnvKey():             fmt.Sprintf("%v", env),
+							}
+							if seDr.SeDnsPrefix != "" && seDr.SeDnsPrefix != common.Default {
+								newServiceEntry.Labels["dnsPrefix"] = seDr.SeDnsPrefix
+							}
```

Comment:
Based on [this](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names) spec for label names, the above might not work. We can have `dns-prefix`?
Also why labels and not an annotations? Are you using this as filter for querying?

Reply:
Good catch will update the name. I don't see use case where we might need to filter base don dns prefix or GTP, will update these to be annotation. 
Will keep env as label, since identity is today getting added and label, and may be required to filter endpoints, let me know if this is ok?

Reply:
Yes

Reply:
cool, updated code based on this